flax/core/scope.py

```diff
@@ -493,15 +460,18 @@ def apply(fn: Callable[..., Any],
     `fn` with the scope partially applied.
   """
   @functools.wraps(fn)
-  def wrapper(variables: FrozenVariables, *args,
-              rngs: Optional[RNGSequences] = None, **kwargs) -> (Any, FrozenVariables):
+  def wrapper(variables: FrozenVariableDict, *args,
+              rngs: Optional[RNGSequences] = None, 
+              **kwargs) -> Union[Any, Tuple[Any, VariableDict]]:
```

Comment:
Are we sure this type signature is correct?  it's not `Tuple[Any, FrozenVariableDict]`?

Reply:
I think it should indeed be `FrozenVariableDict` instead of `VariableDict`, but I think we should use `Union` since depending on whether `mutable` is true or false we will either return only the output (`Any`) or a tuple containing the output and the frozen variables.